library/src/main/java/com/pokegoapi/api/PokemonGo.java

```diff
@@ -108,6 +109,7 @@ public PokemonGo(CredentialProvider credentialProvider, OkHttpClient client, Tim
 		longitude = Double.NaN;
 		latitude = Double.NaN;
 		startTime = currentTimeMillis();
+		seed = playerProfile.getPlayerData().getUsername().length();
```

Comment:
ok, I'm at PC now. This will throw an npe at the first signature generation because getPlayerData is null